File: core/data/src/main/java/me/nalab/core/data/survey/ShortFormQuestion.java
package me.nalab.core.data.survey;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;

@Entity
@Getter
@Setter
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor
public class ShortFormQuestion extends FormQuestion {

	@Enumerated(EnumType.STRING)
	@Column(name = "short_form_question_type")
	private ShortFormQuestionType shortFormQuestionType;

}

File: survey/src/main/kotlin/me/nalab/survey/domain/survey/ShortFormQuestion.kt
package me.nalab.survey.domain.survey

import javax.persistence.*

@Entity
class ShortFormQuestion(
    id: Long? = null,
    title: String,
    order: Int,
    questionType: QuestionType,
    survey: Survey,

    @Enumerated(EnumType.STRING)
    @Column(name = "short_form_question_type")
    private val shortFormQuestionType: ShortFormQuestionType,
) : FormQuestionable(id, title, order, questionType, survey)
